ci/conda_env_cpp.txt

```diff
@@ -31,8 +31,8 @@ gflags
 glog
 gmock>=1.10.0
 google-cloud-cpp>=1.34.0
-grpc-cpp<=1.50.1
 gtest>=1.10.0
+libgrpc
```

Comment:
To clarify why this change is required - the version pin of grpc-cpp here causes an older version of abseil from 2022 to be installed. That version unfortunately appears to have an incorrect pkgconfig file, which does not correctly link in the time_zone shared library and yields missing symbol errors during linkage. This appears to have been fixed in more recent abseil releases

We had a similar issue with the old grpc-cpp package in https://github.com/apache/arrow/pull/46906